File: leet_code/removeDuplicates.js
function removeDuplicates(S, flagged = true) {
  if (!flagged) return S;
  flagged = false;
  let word = S.split('')
    for (let i = 0; i < word.length; i++) {
      if (word[i] === word[i+1]) {
        word[i] = "";
        word[i+1] = "";
        flagged = true;
      }
    }

  return removeDuplicates(word.join(''), flagged);
}


S = "abbaca";
console.log(removeDuplicates(S));

File: leet_code/python/removeDuplicates.py
def removeDuplicates(S, flagged = True):
  print(S)
  if flagged == False: return S

  flagged = False
  word = list(S)
  i = 0
  while i < (len(S) - 1):
    if word[i] == word[i+1]:
      word[i] = ''
      word[i+1] = ''
      flagged = True
      break
    i += 1

  return removeDuplicates(''.join(word), flagged)

S = "abbaca"
removeDuplicates(S)